safe/impact_function/impact_function.py

```diff
@@ -2414,21 +2415,25 @@ def post_process(self, layer):
             layer_title(layer)
 
         for post_processor in post_processors:
-            valid, message = enough_input(layer, post_processor['input'])
-            name = post_processor['name']
-
-            if valid:
-                valid, message = run_single_post_processor(
-                    layer, post_processor)
+            run, run_message = should_run(layer.keywords, post_processor)
+            if run:
+                valid, message = enough_input(layer, post_processor['input'])
+                name = post_processor['name']
                 if valid:
-                    self.set_state_process('post_processor', name)
-                    message = '{name} : Running'.format(name=name)
-                    LOGGER.info(message)
+                    valid, message = run_single_post_processor(
+                        layer, post_processor)
+                    if valid:
+                        self.set_state_process('post_processor', name)
+                        message = '{name} : Running'.format(name=name)
+                        LOGGER.info(message)
 
+                else:
+                    # message = u'{name} : Could not run : {reason}'.format(
```

Comment:
Any reason to comment out this code? I am wondering if it can help when debugging.

Reply:
ah, ok then.